enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,439 @@
+possible to disrupt `PodDisruptionBudgetAtLimit` alert for any user who can
+create PDBs. This should be mitigated by KCM-o ownership of the policy and
+by testing.
+
+### Drawbacks
+
+ValidatingAdmissionPolicy is still in alpha.
+
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Are there any security implications with storing service accounts names in a
+  CRD and comparing these agains `request.userInfo.username` in the
+   `ValidatingAdmissionPolicy` ?
+
+### Test Plan
+
+- e2e test for kube-controller-manager operator should test a presence of
+  the alert and `ValidatingAdmissionPolicy` and that it guards modification
+  of PDBs labels. And correctly excludes labeled PDBs from
+  `PodDisruptionBudgetAtLimit` alert.
+
+### Graduation Criteria
+
+This should not be a user facing API.
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+
+### Version Skew Strategy
+
+### Operational Aspects of API Extensions
+
+
+#### Failure Modes
+
+
+#### Support Procedures
+
+
+## Implementation History
+
+- 2023-01-30: Initial enhancement proposal
+
+
+## Alternatives
+
+### Use a validating admission webhook instead
+
+We could guard a PDB with validating admission webhook to prevent an unhandled
+labels change misuse. Based on the user info from AdmissionRequest we would
+decide if they can modify the label. We could check for either hardcoded
+user/SA that does the request or for KubeVirt alternatively check:
+
+- if the PDB is owned by the VM
+- do SubjectAccessReview if the user has access to the VM
+- check RBAC rights for some KubeVirt resource that would identify a controller
+
+These could be potential issue with this alternative:
+- less reliable than Validating Admission Policy
+- would increase network latency for a core object
+- maintenance burden ( would be probably owned by
+  kube-controller-manager-operator)
+
+### Introduce a new metric
+
+Mainly, KubeVirt should be the only one allowed to disable this alert. We could
+achieve this by introducing a custom metric called for example
+`ocp_kubevirt_vmi_owned_pdbs`. KubeVirt would track its own special PDBs that
+do not conform to the standard use case. We would then subtract this from the
+rest of the PDBs at limit to stop alerting on the KubeVirt ones.
+
+The metric should have at least a namespace and poddisruptionbudget name.
+
+The alert query could look like this:
+
+```json
+max by (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_current_healthy >= kube_poddisruptionbudget_status_desired_healthy and on (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_expected_pods > 0) unless on(namespace, poddisruptionbudget) (ocp_kubevirt_vmi_owned_pdbs))
+```
+
+Kubevirt would maintain/own `ocp_kubevirt_vmi_owned_pdbs` metric and ensure
+only the valid PDBs are ignored. This should prevent users from inserting their
+own PDBs into this ignore list.
```

Comment:
I'm not aware of a mechanism where one ServiceMonitor (e.g. KubeVirt's) can forbid other ServiceMonitors from serving a given metric.  I guess monitoring only scrapes ServiceMonitors from namespaces with `openshift.io/cluster-monitoring: "true"`, and that limits the exposure.  But I'm not sure what would guard against a flow like:

1. User creates a new project.
2. User adds the `openshift.io/cluster-monitoring: "true"` label to their project.
3. User creates a ServiceMonitor asking to be scraped.
4. User serves `ocp_kubevirt_vmi_owned_pdbs` out of their ServiceMonitor with `namespace` and `poddisruptionbudget` labels matching many well-known PDBs.
5. `PodDisruptionBudgetAtLimit` doesn't fire for those PDBs, even when the cluster administrator wishes it  would.

Maybe there are guards against labeling namespaces?  But spot-checking a random CI namespace in build02:

```console
$ oc --as system:admin policy who-can -o json patch namespace ci-op-whbx73iy | jq '.users | length'
56
```

That's a lot of folks with access, even if most of them are `system:serviceaccount:openshift-*`.  Are there other guards securing this pathway?